shesha-reactjs/src/designer-components/collapsiblePanel/collapsiblePanelComponent.tsx

```diff
@@ -54,39 +60,54 @@ const CollapsiblePanelComponent: IToolboxComponent<ICollapsiblePanelComponentPro
     };
 
     const headerComponents = model?.header?.components ?? [];
+
+    const hasCustomHeader = model?.hasCustomHeader;
+
     const extra =
-      headerComponents?.length > 0 || formMode === 'designer' ? (
+      ((headerComponents?.length > 0 || formMode === 'designer') && !hasCustomHeader) ? (
         <ComponentsContainer
           containerId={model.header?.id}
           direction="horizontal"
-          dynamicComponents={model?.isDynamic ? headerComponents : []}
+          dynamicComponents={model?.isDynamic ? model.header?.components : []}
         />
       ) : null;
 
+    const panelPosition = !!panelContextState ? 'child' : 'parent';
+
     return (
       <ParentProvider model={model}>
-        <CollapsiblePanel
-          header={evaluatedLabel}
-          expandIconPosition={expandIconPosition !== 'hide' ? (expandIconPosition as ExpandIconPosition) : 'start'}
-          collapsedByDefault={collapsedByDefault}
-          extra={extra}
-          collapsible={collapsible === 'header' ? 'header' : 'icon'}
-          showArrow={collapsible !== 'disabled' && expandIconPosition !== 'hide'}
-          ghost={ghost}
-          dynamicBorderRadius={model?.borderRadius}
-          style={{...getPanelStyle}}
-          className={model.className}
-          bodyColor={bodyColor}
-          headerColor={headerColor}
-          isSimpleDesign={isSimpleDesign}
-          hideCollapseContent={hideCollapseContent}
-          hideWhenEmpty={hideWhenEmpty}
-        >
-          <ComponentsContainer
-            containerId={model.content.id}
-            dynamicComponents={model?.isDynamic ? model?.content.components : []}
-          />
-        </CollapsiblePanel>
+        <PanelContext.Provider value={panelPosition}>
+          <CollapsiblePanel
+            header={hasCustomHeader ?
+              <ComponentsContainer
+                containerId={model.customHeader.id}
+                dynamicComponents={(model?.isDynamic) ? model?.customHeader?.components : []}
+              /> :
+              evaluatedLabel
+            }
+            expandIconPosition={expandIconPosition !== 'hide' ? (expandIconPosition as ExpandIconPosition) : 'start'}
+            collapsedByDefault={collapsedByDefault}
+            extra={extra}
+            collapsible={collapsible === 'header' ? 'header' : 'icon'}
+            showArrow={collapsible !== 'disabled' && expandIconPosition !== 'hide'}
+            ghost={ghost}
+            dynamicBorderRadius={model?.borderRadius}
+            style={{ ...getPanelStyle }}
+            className={model.className}
+            bodyColor={bodyColor}
+            headerColor={headerColor}
+            isSimpleDesign={isSimpleDesign}
+            parentPanel={panelPosition === 'parent'}
+            hideCollapseContent={hideCollapseContent}
+            hideWhenEmpty={hideWhenEmpty}
+            isSettingPanel={!model?.componentName}
```

Comment:
Don't ever do that!

It is completely unclear how the is Setting Panel property is related to Component Name. The configurator can specify Component Name and then your logic will break. You can't make such non-obvious relations!

To determine that the form is in settings mode, you can do the following:
`const form = useForm();`
...
`isSettingPanel={form.formSettings.isSettingsForm}`

In addition, it is completely wrong to add a form-specific settings mode to CollapsiblePanel, since this is a common component and should not be associated with the Form Designer in any way.

The correct solution would be to add some property, for example, Default Header Border or something, which will set the default header type.

Or сustomHeaderBorder with the ability to specify a specific style